### ISSUE TITLE
`quoted-annotation` (`UP037`), `undefined-name` (`F821`) & `quoted-type-alias` (`TC008`) - false positives on pyright inlined `TypedDict`s

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#10812]
pyright supports defining a `TypedDict` inline like so:

```py
from __future__ import annotations
from typing import TypedDict

foo: TypedDict[{"bar": str}] = {"bar": "bar"} # errors: F821, UP037
```

removing the quotes causes the following pyright error:

```py
foo: TypedDict[{bar: str}] = {"bar": "bar"} # error: Expected string literal for dictionary entry name
```
[playground](https://play.ruff.rs/ad657c55-d2c6-4be5-ade7-63212003dba0)

**Comment by AlexWaygood** [comment]
Thanks, I agree that ruff's behaviour is incorrect here.

For other readers who may not be familiar — this is an experimental syntax for "inline TypedDicts" that has been floated on the (now-defunct) typing-sig mailing list. Pyright has added support for this experimental syntax so that people can experiment with it. Some more details here: https://github.com/python/typing/discussions/1391.

I don't believe a PEP has yet been written to formally propose this new way of writing TypedDicts. But without this new syntax, it wouldn't be a valid PEP-484 type annotation anyway, so regardless I think ruff probably shouldn't be telling people to remove the quotes in that type annotation.

**Comment by DetachHead** [comment]
fyi there's now a draft PEP for this: https://github.com/python/peps/pull/4082

**Comment by DetachHead** [comment]
also happens with the new `quoted-type-alias` (`TC008`) rule in 0.8.1:

```py
from typing import TypedDict

type Foo = TypedDict[{"Bar": str}] # error: Remove quotes from type alias
```
